File: DataProcessingSystemJava/src/main/java/com/example/dataprocessing/service/TaskQueue.java
// TaskQueue.java
package com.example.dataprocessing.service;

import com.example.dataprocessing.model.Task;
import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;

public class TaskQueue {
    private final BlockingQueue<Task> queue = new LinkedBlockingQueue<>();

    public void addTask(Task task) throws InterruptedException {
        queue.put(task);
    }

    public Task getTask() throws InterruptedException {
        return queue.take();
    }

    public boolean isEmpty() {
        return queue.isEmpty();
    }
}

File: GoDataProcessing/task_queue.go
package main

type TaskQueue struct {
	tasks chan Task
}

func NewTaskQueue(size int) *TaskQueue {
	return &TaskQueue{tasks: make(chan Task, size)}
}

func (q *TaskQueue) AddTask(task Task) {
	q.tasks <- task
}

func (q *TaskQueue) GetTask() <-chan Task {
	return q.tasks
}

func (q *TaskQueue) Close() {
	close(q.tasks)
}
